### PR TITLE
Collector + directory doc fixes

### PR DESCRIPTION
**Pull request by saroh**
Just fixing a few links :) 

#1232

### CODE REVIEW

File: src/directory/directory.rs [hunk 1]
```diff
@@ -43,7 +43,6 @@ impl RetryPolicy {
 }
 
 /// The `DirectoryLock` is an object that represents a file lock.
-/// See  [`LockType`](struct.LockType.html)
```
Comment (saroh) [inline src/directory/directory.rs:46]: `DirectoryLockGuard` ? It's private though.

Comment (fulmicoton) [reply src/directory/directory.rs:46]: No clue what it was

**Review by fulmicoton — APPROVED** [review]
Thank you @saroh 
Do you know if there is a command to check for broken links in the rustdoc?
thtat would be cool to add that to the CI

**Comment by saroh** [comment]
> Do you know if there is a command to check for broken links in the rustdoc?

@fulmicoton seems supported https://doc.rust-lang.org/rustdoc/lints.html I'll give it a go for the future packages :)

**Comment by saroh** [comment]
https://github.com/saroh/tantivy/commit/6ef8e20c122a7c49ff79cdb3bab3acd503d584cc should make `cargo doc` fail providing it's in your CI 
the doc passes as is.

**Pull request merged**